src/main/resources/db_scripts/postgres-rutherford-create-script.sql

```diff
@@ -695,6 +695,8 @@ CREATE TABLE public.users (
     school_id text,
     school_other text,
     exam_board text,
+    registered_contexts jsonb[] DEFAULT array[]::jsonb[] NOT NULL,
```

Comment:
I was aware that jsonb handled arrays at the top level, but I went with array because postgres knowing it is an array means it has extra ways to query and aggreagate accross it. Not sure if we'd ever need those types of queries but I thought it was worth the extra line or two of set up. It is definitely odd at first sight though so maybe it's not worth the confusion.

Reply:
I did some Googling and at least one person on the internet agrees with your method:
https://stackoverflow.com/a/56625335
and suggests it is (for our volume although irrelevant for our current use-case) faster to use arrays, and it is trivial to convert from JSON array to ARRAY arrays and back.
So I guess stick with it since it's done!

It'll need an exclusive lock on the table to initialise it to an empty array by default, so adding the column might queue queries for several minutes!

Reply:
I thought it didn't need an exclusive lock for default values in more recent versions of Postgres?

Reply:
I think they've basically implemented a default feedback option like us but for queries to a column! :)

Reply:
https://www.postgresql.org/docs/13/sql-altertable.html

>When a column is added with ADD COLUMN and a non-volatile DEFAULT is specified, the default is evaluated at the time of the statement and the result stored in the table's metadata. That value will be used for the column for all existing rows. If no DEFAULT is specified, NULL is used. In neither case is a rewrite of the table required.

Reply:
Oh that's neat! I'm glad we upgraded to use 11 (and on to 13) where it first arrived. I'd missed that change!